app/permissions_migrations.go

```diff
@@ -74,18 +75,66 @@ const (
 	PermissionManageRemoteClusters           = "manage_remote_clusters" // deprecated; use `manage_secure_connections`
 )
 
-func isRole(roleName string) func(*model.Role, map[string]map[string]bool) bool {
+// Deprecated: This function should only be used if a case arises where team and/or channel scheme roles do not need to be migrated.
+// Otherwise, use isRole.
+func isExactRole(roleName string) func(*model.Role, map[string]map[string]bool) bool {
 	return func(role *model.Role, permissionsMap map[string]map[string]bool) bool {
 		return role.Name == roleName
 	}
 }
 
+// isRole returns true if roleName matches a role's name field or if the a team
+// or channel scheme role matches a "common name". A common name is one of the following role
+// that is common among the system scheme and the team and/or channel schemes:
+//
+// TeamAdmin,
+// TeamUser,
+// TeamGuest,
+// ChannelAdmin,
+// ChannelUser,
+// ChannelGuest,
+// PlaybookAdmin,
+// PlaybookMember,
+// RunAdmin,
+// RunMember
+func isRole(roleName string) func(*model.Role, map[string]map[string]bool) bool {
+	return func(role *model.Role, permissionsMap map[string]map[string]bool) bool {
+		if role.Name == roleName {
+			return true
+		}
+		return isSchemeRoleAssociatedToCommonName(roleName, role)
+	}
+}
+
 func isNotRole(roleName string) func(*model.Role, map[string]map[string]bool) bool {
```

Comment:
Should `isNotRole` similarly use `isSchemeRoleAssociatedToCommonName`?

Reply:
I didn't think about `isNotRole`. What I've done is renamed the implemented and implemented a new `isNotRole` function. Let me know if you think that works.